libs/src/api/user.js

```diff
@@ -596,6 +604,32 @@ class Users extends Base {
     }
   }
 
+  /**
+   * If a user's creator_node_endpoint is null:
+   * 1. Generate a replica set
+   * 2. Use the `upgradeToCreator` method to
+   *    - Update the replica set
+   *    - Sync user data over from UM if any
+   * Used during the sanity check and in uploadImage() in files.js
+   */
+  async assignReplicaSetIfNecessary () {
+    const user = this.userStateManager.getCurrentUser()
+
+    // If no user is logged in, or a creator node endpoint is already assigned,
+    // skip this call
+    if (!user || user.creator_node_endpoint) return
+
+    // Generate a replica set and assign to user
+    try {
+      const { primary, secondaries } = await this.ServiceProvider.autoSelectCreatorNodes({})
+      const contentNodeEndpoint = CreatorNode.buildEndpoint(primary, secondaries)
+      const currentEndpoint = this.userStateManager.getCurrentUser().creator_node_endpoint
+      await this.upgradeToCreator(currentEndpoint, contentNodeEndpoint, false)
```

Comment:
why call this vs assignReplicaSet @vicky-g

Reply:
`upgradeToCreator()` also syncs data from UM to the assigned replica set. this is for the users that have currently don't have a CNE, and have data on UM. for freshly signed up users who shouldn't have any data on UM, this will not be necessary.

Reply:
discussed offline -- add UM as the explicit `currentEndpoint`